tests/phpunit/integration/tests/REST_API/Stories_Taxonomies_Controller.php

```diff
@@ -95,6 +95,27 @@ public function test_get_items_hierarchical_false(): void {
 		}
 	}
 
+	/**
+	 * @covers ::get_items
+	 * @covers ::get_collection_params
+	 */
+	public function test_get_items_show_ui_false(): void {
+		$this->controller->register();
+
+		$request = new WP_REST_Request( WP_REST_Server::READABLE, '/web-stories/v1/taxonomies' );
+		$request->set_param( 'show_ui', false );
+		$request->set_param( 'type', 'attachment' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertFalse( $response->is_error() );
+		$data = $response->get_data();
+		$this->assertNotEmpty( $data );
+		$this->assertCount( 1, $data );
+		$this->assertArrayHasKey( 'web_story_media_source', $data );
+		$this->assertArrayHasKey( 'name', $data['web_story_media_source'] );
+		$this->assertSame( 'Source', $data['web_story_media_source']['name'] );
```

Comment:
While the feature works and the test passes, the test does not actually verify that the taxonomy has `show_ui` set to `false`

Assertions like this would be more robbust:

```suggestion
		$this->assertFalse( $response->is_error() );
		$data = $response->get_data();
		$this->assertNotEmpty( $data );
		foreach( array_keys( $data ) as $taxonomy ) {
			$tax = get_taxonomy( $taxonomy );
			$this->assertInstanceOf(\WP_Taxonomy::class, $tax);
			$this->assertFalse($tax->show_ui);
		}
```

In addition to that, it would be great to have tests for the `show_ui=>true` case as well. Compare this to the tests for the `hierarchical` option, where all three cases are covered:

* `test_get_items`: covers default case (all taxonomies included)
* `test_get_items_hierarchical_false`
* `test_get_items_hierarchical_true`

It should be easy to copy these tests and adapt them for `show_ui`

Reply:
Refactored the test to be a little cleaner.

Reply:
I take it you see little benefit in adding tests for the other cases (default, false, true)?

Reply:
> I take it you see little benefit in adding tests for the other cases (default, false, true)?

Correct, but I can add it if you want.

Reply:
I have updated tests to make them a lot cleaner.